libs/dh/shared/feature-authorization/src/lib/actor-token.service.ts

```diff
@@ -15,48 +15,118 @@
  * limitations under the License.
  */
 
+import {
+  HttpEvent,
+  HttpHandler,
+  HttpRequest,
+  HttpResponse,
+} from '@angular/common/http';
 import { Injectable } from '@angular/core';
 import {
   MarketParticipantUserHttp,
   TokenHttp,
 } from '@energinet-datahub/dh/shared/domain';
 import { map, Observable, of, switchMap, tap } from 'rxjs';
 
+type CachedEntry = { token: string; value: string };
+
 @Injectable({ providedIn: 'root' })
 export class ActorTokenService {
-  private _internalToken = '';
-  private _externalToken = '';
+  private _internalActors: CachedEntry = {
+    token: '',
+    value: '',
+  };
+
+  private _internalToken: CachedEntry = {
+    token: '',
+    value: '',
+  };
 
   constructor(
     private marketParticipantUserHttp: MarketParticipantUserHttp,
     private tokenHttp: TokenHttp
   ) {}
 
-  public acquireToken = (externalToken: string): Observable<string> => {
-    if (this._externalToken !== externalToken) {
-      this._internalToken = '';
-    }
+  public isPartOfAuthFlow(request: HttpRequest<unknown>) {
+    return this.isUserActorsRequest(request) || this.isTokenRequest(request);
+  }
 
-    this._externalToken = externalToken;
+  public handleAuthFlow(
+    request: HttpRequest<unknown>,
+    nextHandler: HttpHandler
+  ): Observable<HttpEvent<unknown>> {
+    if (this.isUserActorsRequest(request)) {
+      return this.updateCache(
+        request,
+        nextHandler,
+        () => this._internalActors,
+        (value) => (this._internalActors = value)
+      );
+    }
 
-    if (this._internalToken) {
-      return of(this._internalToken);
+    if (this.isTokenRequest(request)) {
+      return this.updateCache(
+        request,
+        nextHandler,
+        () => this._internalToken,
+        (value) => (this._internalToken = value)
+      );
     }
 
+    return nextHandler.handle(request);
+  }
+
+  public acquireToken = (): Observable<string> => {
     return this.marketParticipantUserHttp
-      .v1MarketParticipantUserGet(externalToken)
+      .v1MarketParticipantUserActorsGet()
       .pipe(
         switchMap((r) => {
           return this.tokenHttp
-            .v1TokenPost({
-              actorId: r.actorIds[0],
-              externalToken: externalToken,
-            })
-            .pipe(
-              tap((r) => (this._internalToken = r.token)),
-              map((r) => r.token)
-            );
+            .v1TokenPost(r.actorIds[0])
```

Comment:
question: is the first id always the right one? If yes, is there a reason for returning an array from the endpoint?

Reply:
Yes, it is for now. Next feature (Feature 3) will add the drop-up to allow for changing this value.